python/pyspark/sql/functions.py

```diff
@@ -2326,7 +2326,7 @@ def schema_of_json(json):
     >>> df.select(schema_of_json('{"a": 0}').alias("json")).collect()
     [Row(json=u'struct<a:bigint>')]
     """
-    if isinstance(json, basestring):
+    if isinstance(json, str):
```

Comment:
The problem here is we will not support unicode in Python 2 ..

Reply:
shall we apply it to 2.4? I'm not aware of the background, why we did not put
```
if sys.version >= '3':
    basestring = str
```
in 2.4?